src/EditorFeatures/CSharpTest/ConvertAnonymousTypeToClass/ConvertAnonymousTypeToClassTests.cs

```diff
@@ -1399,6 +1399,126 @@ void Method()
     }
 }
 
+internal class NewClass
+{
+    public int A { get; }
+    public int B { get; }
+
+    public NewClass(int a, int b)
+    {
+        A = a;
+        B = b;
+    }
+
+    public override bool Equals(object obj)
+    {
+        return obj is NewClass other &&
+               A == other.A &&
+               B == other.B;
+    }
+
+    public override int GetHashCode()
+    {
+        var hashCode = -1817952719;
+        hashCode = hashCode * -1521134295 + A.GetHashCode();
+        hashCode = hashCode * -1521134295 + B.GetHashCode();
+        return hashCode;
+    }
+}";
+            await TestInRegularAndScriptAsync(text, expected, options: this.PreferImplicitTypeWithInfo());
+        }
+
+        [WorkItem(45747, "https://github.com/dotnet/roslyn/issues/45747")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertAnonymousTypeToClass)]
+        public async Task ConvertOmittingTrailingComma()
+        {
+            var text = @"
+class Test
+{
+    void Method()
+    {
+        var t1 = [||]new
+        {
+            a = 1,
+            b = 2,
+        };
+    }
+}
+";
+            var expected = @"
+class Test
+{
+    void Method()
+    {
+        var t1 = new {|Rename:NewClass|}(
+1,
+2
+        );
```

Comment:
```suggestion
2
        );
```

this doesn't seem desirable.  If there is only whitespace trivia, i think it would be appropriate to just not transfer that along.

Reply:
Do you want to change the behavior in the no-trailing-comma case as well? Right now
```
            var t1 = new
            {
                a = 1,
                b = 2
            };
```
is transformed into
```
            var t1 = new NewClass(
1,
2
            );
```
Also, other refactorings (like "Convert to tuple") have similar behavior, and I feel there should be consistency.

Reply:
i think there is a distinct difference int he patterns used when using `{` style versus (an initialization list) vs `(` style (an argument list).  I would prefer we matched the end style we expect people to most commonly use.  That said, i'm ok with this being out of scope for this PR as we'd likely need to update a few things at the same time for consistency :)

Reply:
Agreed, I'll create an issue for that.

Reply:
Created issue #45792